### PR TITLE
Add pixel setters and getters for DisplayFrame

### PR DESCRIPTION
**Pull request by Nycto**
Allows for individual pixels to be read and set on the display frame

Based on the code found here:
https://devforum.play.date/t/c-macros-for-working-with-playdate-bitmap-data/7706

### CODE REVIEW

File: src/playdate/graphics.nim [hunk 1]
```diff
@@ -243,6 +243,30 @@ proc getDisplayFrame*(this: ptr PlaydateGraphics): DisplayFrame =
     privateAccess(PlaydateGraphics)
     return cast[DisplayFrame](this.getDisplayFrame()) # Who should manage this memory? Not clear. Not auto-managed right now.
 
+proc frameIndex(x, y: SomeInteger): int = y * LCD_ROWSIZE + x div 8
+    ## Returns the index of a coordinate within a DisplayFrame
```
Comment (samdze) [inline src/playdate/graphics.nim]: Documentation should end with a period.
I'll create a new issue, writing proper documentation for the entire module would be nice

File: src/playdate/graphics.nim [hunk 2]
```diff
@@ -243,6 +243,30 @@ proc getDisplayFrame*(this: ptr PlaydateGraphics): DisplayFrame =
...
+
+proc set*(frame: DisplayFrame, x, y: SomeInteger) =
+    ## Sets the pixel at x, y to black
+    frame[frameIndex(x, y)] = frame[frameIndex(x, y)] and not frameBit(x)
+
+proc clear*(frame: DisplayFrame, x, y: SomeInteger) =
+    ## Clears the color from a pixel at the given coordinate
+    frame[frameIndex(x, y)] = frame[frameIndex(x, y)] or frameBit(x)
+
+proc `[]=`*(frame: DisplayFrame, x, y: SomeInteger, color: LCDSolidColor) =
```
Comment (samdze) [inline src/playdate/graphics.nim]: I think the square brackets operator is ok, but is there a way to show it on the autocomplete list on vscode?
Seems like there isn't.

File: src/playdate/graphics.nim [hunk 3]
```diff
@@ -243,6 +243,30 @@ proc getDisplayFrame*(this: ptr PlaydateGraphics): DisplayFrame =
     privateAccess(PlaydateGraphics)
     return cast[DisplayFrame](this.getDisplayFrame()) # Who should manage this memory? Not clear. Not auto-managed right now.
 
+proc frameIndex(x, y: SomeInteger): int = y * LCD_ROWSIZE + x div 8
+    ## Returns the index of a coordinate within a DisplayFrame
+
+proc frameBit(x: SomeInteger): uint8 = 1'u8 shl uint8(7 - (x mod 8))
+
+proc `[]`*(frame: DisplayFrame, x, y: SomeInteger): LCDSolidColor =
```
Comment (samdze) [inline src/playdate/graphics.nim]: Using generic types like SomeInteger is fine, but I think it would be better to have them as a separate PR that changes the signature all around the module, where feasible.

Also, there's a small issue in VSCode using this type:
Code 2023-01-31 at 12 57 16

That \*missing parameters\* in the proc signature appears when using generic types.

**Review by samdze — CHANGES_REQUESTED** [review]

File: src/playdate/graphics.nim (hunk 3 above)
Comment (Nycto) [reply src/playdate/graphics.nim]: I'm fine with removing them for now

File: src/playdate/graphics.nim (hunk 2 above)
Comment (Nycto) [reply src/playdate/graphics.nim]: I don't think there is. We can always rename this to `set` instead

**Comment by Nycto** [comment]
Pushed an update. I also added bounds checks as it started throwing exceptions when setting pixels outside the range of the display frame.

**Comment by samdze** [comment]
Thank you!
